README.md

```diff
@@ -4,32 +4,123 @@
+- Intel Visual Compute Accelerator (Intel VCA)
+
+The demo subdirectory contains both a [GPU plugin demo video](demo#intel-gpu-device-plugin-demo-video)
+as well as code for an OpenCL [FFT demo](demo/ubuntu-demo-opencl).
+
+### FPGA device plugin
+
+The [FPGA device plugin](cmd/fpga_plugin/README.md) supports Intel Arria 10 and Stratix 10
+FPGA passthrough. It supports the [OPAE](https://01.org/opae) mode.
+
+The FPGA plugin comes as three parts. Start with the [device plugin](#device-plugin), and then
+examine the [admissing controller](#admission-controller) and the
+[CRIO-O prestart hook](#cri-o-prestart-hook).
+
+> **FIXME** - unclear where the [fpga_tool](cmd/fpga_tool) comes into play. We should document this.
+
+The demo subdirectory contains a [video](demo#intel-fpga-device-plugin-demo-video) showing deployment
+and use of the FPGA plugin. Sources relating to the demo can be found in the
+[opae-nlb-demo](demo/opae-nlb-demo) subdirectory.
+
+#### device plugin
+
+The [FPGA device plugin](cmd/fpga_plugin/README.md) is responsible for discovering and reporting FPGA
+devices to `kubelet`. During the allocation phase, it instructs `kubelet` about device nodes to be
+accessed by the container and sets the annotation that is later on passed to CRI to trigger programming
+by the CRI-O hook.
+
+#### admission controller
+
+The [FPGA admission controller webhook](cmd/fpga_admissionwebhook/README.md) is responsible for performing
+mapping from user-friendly function IDs to the Interface ID and Bitstream ID that are required for FPGA
+programming by the CRI-O hook. Mappings are stored in namespaced custom resource definition (CRD) objects,
+therefore the admission controller also performs access control, determining which bitstream can be used
+for which namespace. The admission controller also keeps the user from bypassing namespaced mapping
+restrictions, by denying admission of any pods that are trying to use internal knowledge of InterfaceID
+or Bitstream ID environment variables used by the prestart hook. 
+
+#### CRI-O prestart hook
+
+The [FPGA prestart CRI-O hook](cmd/fpga_crihook/README.md) is triggered by pod annotation set by the
+FPGA device plugin. It performs discovery of the requested FPGA function bitstream and then programs
+FPGA devices based on the environment variables in the workload description.
+
+### [QAT](https://01.org/intel-quick-assist-technology) device plugin
+
+The [QAT plugin](cmd/qat_plugin/README.md) supports device plugin for Intel QAT adapters, and includes
+code [showing deployment](cmd/qat_plugin/dpdkdrv) via [DPDK](https://doc.dpdk.org/guides/cryptodevs/qat.html).
+
+the demo subdirectory includes details of both a
+[QAT DPDK demo](demo#intel-quickassist-technology-device-plugin-with-dpdk-demo-video)
+and a [QAT OpenSSL demo](demo#intel-quickassist-technology-device-plugin-openssl-demo-video).
+Source for the OpenSSL demo can be found in the [relevant subdirectory](demo/openssl-qat-engine).
```

Comment:
Perhaps link https://github.com/kata-containers/documentation/blob/master/use-cases/using-Intel-QAT-and-kata.md
?

Reply:
Ack. adding. added in the QAT section rather than the 'related code' section so it is more obvious to folks interested in QAT.